src/hotspot/share/prims/scopedMemoryAccess.hpp

```diff
@@ -29,7 +29,7 @@
 #include "jni.h"
 
 extern "C" {
-  void JNICALL JVM_RegisterJDKInternalMiscScopedMemoryAccessMethods(JNIEnv *env, jobject rec, jobject scope, jthrowable exception);
+  void JNICALL JVM_RegisterJDKInternalMiscScopedMemoryAccessMethods(JNIEnv *env, jclass scopedMemoryAccessClass);
```

Comment:
Why isn't this declared with JNIEXPORT like the functions in jvm.h?

Reply:
It's similar to the declarations in nativeLookup.cpp. Those don't have JNIEXPORT either.

FWIW, this symbol doesn't need to/shouldn't be exported, since we only reference it internally to pass the function pointer to a call to RegisterNatives, and it's not meant to be called directly when an external program links against libjvm.

The functions in jvm.h are part of an API though, so those need to be exported. e.g. on Windows the macro expands to __declspec(dllexport), which is used to make the symbol externally visible in the final jvm.dll.